File: tennis-server/src/main/java/com/sa/service/GameService.java
package com.sa.service;

import com.sa.model.Game;
import com.sa.model.Player;
import com.sa.model.SetTennis;

public interface GameService {

	public Game createGame(SetTennis setTennis);
	public Game saveGame(Game game);
	public void addPointGame(Game game, Player player);
	public Game addPointGame(Long gameId, Long playerId);
	public Game getGameById(Long gameId);
	public Game getCurrentGame(Long setTennisId);
	public Game getLastGame(Long setTennisId);
	public int getGameOrder(Game game);
}

File: tennis-web/src/app/service/game.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { PlayerDto } from './player.service';

export class GameDto {
    constructor(
        public id: string,
        public scores: ScoreGameDto[],
        public scoresMap: Map<string, ScoreGameDto []>,
        public players: PlayerDto [],
        public winner: PlayerDto,
        public gameOrder: string,
    ) { }
}

export class ScoreGameDto {
    constructor(
        public id: string,
        public scoreValue: string,
        public player: PlayerDto
    ) { }
}

@Injectable({
  providedIn: 'root'
})
export class GameService {

  constructor(private httpClient: HttpClient) { }

   getCurrentGame(setTennisId: string) {
      return this.httpClient.get<GameDto>('http://localhost:8080/game/currentGame' + '/' + setTennisId);
  }

  addPoint(gameId: string, playerId: string) {
      return this.httpClient.get<GameDto>('http://localhost:8080/game/addPoint' + '/' + gameId + '/' + playerId);
  }

  createGame(setTennisId: string) {
      return this.httpClient.get<GameDto>('http://localhost:8080/game/createGame' + '/' + setTennisId);
  }
}
